integration/control_plane.go

```diff
@@ -10,10 +10,15 @@ import (
 // Right now, that means Etcd and your APIServer. This is likely to increase in
 // future.
 type ControlPlane struct {
-	APIServer         *APIServer
-	Etcd              *Etcd
-	ControllerManager *ControllerManager
-	Scheduler         *Scheduler
+	APIServer            *APIServer
+	Etcd                 *Etcd
+	AdditionalComponents []ControlPlaneComponent
```

Comment:
I'm not sure yet what I think about that :-)

Reply:
I understand. My reasoning here:
- Etcd is a completely different thing -- that should be its own type
- APIServer is also "special" and the central component
- All the other components seem to be very same thing:
  * they need to know where the APIServer is
  * they can be started and stopped independently

Therefor I encapsulated that in this interface.